### PR TITLE
Update the compatibility matrix

### PR DESCRIPTION
**Pull request by timoreimann**
**What this PR does / why we need it**:
Update the compatibility matrix.

Improve formatting along the way to satisfy Github's markdown rules, which helps with rendering the tables outside of Github (e.g., Goland IDE).

Based on [this discussion on Slack](https://kubernetes.slack.com/archives/CTND9FHU2/p1638524022013200).

```release-note
NONE
```

/assign @cpanato

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,21 +40,28 @@ Check out the [getting started guide](./docs/getting-started.md) for launching a
...
+| Kubernetes 1.15 | ✓                                     | ✓                                     |                                       |                                       |                                       |                                 |
+| Kubernetes 1.16 |                                       |                                       | ✓                                     |                                       |                                       |                                 |
+| Kubernetes 1.17 |                                       |                                       | ✓                                     | ✓                                     |                                       |                                 |
+| Kubernetes 1.18 |                                       |                                       |                                       | ✓                                     |                                       | ✓(?)                            |
```
Comment (timoreimann) [inline README.md:61]: The colums with a `(?)` need special review attention as I wasn't certain about our compatibility.

File: README.md [hunk 2]
```diff
@@ -40,21 +40,28 @@ Check out the [getting started guide](./docs/getting-started.md) for launching a
...
+| Kubernetes 1.19 |                                       |                                       |                                       | ✓                                     |                                       | ✓(?)                            |
+| Kubernetes 1.20 |                                       |                                       |                                       | ?                                     |                                       | ✓(?)                            |
+| Kubernetes 1.21 |                                       |                                       |                                       | ?                                     | ?                                     | ✓(?)                            |
```
Comment (cpanato) [inline README.md:65]: you can add 1.22 and 1.23 as well both works in the 1.0
also v1 should work in 1.19/1.20 as well

not sure in 1.18 but that is already deprecated so we can remove that out

**Review by cpanato — COMMENTED** [review]
thanks for working on this, just a small comment

**Comment by timoreimann** [comment]
@cpanato I dropped Kubernetes versions 1.18 and below from the table, which I think is what you suggested (but let me know if I misunderstood).

For the v0.x support with regards to the remaining Kubernetes releases, I checked how far [the Cluster API compatibility](https://cluster-api.sigs.k8s.io/reference/versions.html) goes and tried to match it with our table. Could you double check to confirm whether everything looks correct now?

Thanks!

**Review by cpanato — APPROVED** [review]
looks great thanks for the update
/lgtm
/approve

**Pull request merged**